flink-state-backends/flink-statebackend-changelog/src/test/java/org/apache/flink/state/changelog/ChangelogStateBackendTestUtils.java

```diff
@@ -76,4 +120,145 @@ public static TestTaskStateManager createTaskStateManager(File changelogStorageP
                                 Path.fromLocalFile(changelogStoragePath), false, 1024))
                 .build();
     }
+
+    public static void testMaterializedRestore(
+            StateBackend stateBackend, Environment env, CheckpointStreamFactory streamFactory)
+            throws Exception {
+        SharedStateRegistry sharedStateRegistry = new SharedStateRegistry();
+
+        TypeInformation<StateBackendTestBase.TestPojo> pojoType =
+                new GenericTypeInfo<>(StateBackendTestBase.TestPojo.class);
+        ValueStateDescriptor<StateBackendTestBase.TestPojo> kvId =
+                new ValueStateDescriptor<>("id", pojoType);
+
+        ChangelogKeyedStateBackend<Integer> keyedBackend =
+                (ChangelogKeyedStateBackend<Integer>) createKeyedBackend(stateBackend, env);
+
+        CompletableFuture<Void> asyncComplete = new CompletableFuture<>();
+        PeriodicMaterializationManager periodicMaterializationManager =
+                new PeriodicMaterializationManager(
+                        checkNotNull(env.getMainMailboxExecutor()),
+                        checkNotNull(env.getAsyncOperationsThreadPool()),
+                        env.getTaskInfo().getTaskNameWithSubtasks(),
+                        (message, exception) -> asyncComplete.completeExceptionally(exception),
+                        keyedBackend,
+                        10,
+                        1);
+
+        periodicMaterializationManager.start();
+
+        try {
+            ValueState<StateBackendTestBase.TestPojo> state =
+                    keyedBackend.getPartitionedState(
+                            VoidNamespace.INSTANCE, VoidNamespaceSerializer.INSTANCE, kvId);
+
+            keyedBackend.setCurrentKey(1);
+            state.update(new StateBackendTestBase.TestPojo("u1", 1));
+
+            keyedBackend.setCurrentKey(2);
+            state.update(new StateBackendTestBase.TestPojo("u2", 2));
+
+            Thread.sleep(100);
```

Comment:
I'm assuming this part isn't finished yet. 
Otherwise, this will make the test unstable.